app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalController.php

```diff
@@ -84,4 +85,32 @@ public function show(): JsonResponse
             'footer_logo' => Vite::asset('resources/svg/CGBS_Logo_FullColor_Light.svg'),
         ]);
     }
+
+    public function getProductLicenses(): JsonResponse
+    {
+        $licenses = auth('contact')->user()->productLicenses()
+            ->with(['product:id,name,version'])
+            ->get()
+            ->map(function ($license) {
+                $license->formatted_expiration_date = $license->expiration_date
+                    ? $license->expiration_date->format('m-d-Y')
+                    : null;
+
+                return $license;
+            });
+
+        $activeLicenses = $licenses->filter(function ($license) {
+            return $license->status === ProductLicenseStatus::Active;
+        });
+
+        $expiredLicenses = $licenses->filter(function ($license) {
+            return $license->status === ProductLicenseStatus::Expired;
+        });
+
+        return response()->json([
+            'success' => true,
+            'activeLicense' => $activeLicenses->values(),
+            'expiredLicense' => $expiredLicenses->values(),
+        ]);
```

Comment:
Rather than defining this in variables it could be cleaner here to just inline them into the array.